From: Thomas Reitmayr <treitmayr@devbase.at>
Date: Mon, 29 Dec 2008 13:59:29 +0000 (-0300)
Subject: V4L/DVB (9981): [PATCH] usb-urb.c: Fix initialization of URB list.
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=d29ca09d50309bb7df5d57aec88efdcd09a56f9f

V4L/DVB (9981): [PATCH] usb-urb.c: Fix initialization of URB list.

Fix the initialization of the URB list for a DVB-USB device to prevent
problems on certain platforms (MIPS).

Signed-off-by: Thomas Reitmayr <treitmayr@devbase.at>
Signed-off-by: Patrick Boettcher <pb@linuxtv.org>
Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>
---

--- a/drivers/media/dvb/dvb-usb/usb-urb.c	2008-12-31 09:48:53.000000000 +0000
+++ b/drivers/media/dvb/dvb-usb/usb-urb.c	2008-12-31 09:48:58.000000000 +0000
@@ -152,7 +152,8 @@
 				stream->props.u.bulk.buffersize,
 				usb_urb_complete, stream);
 
-		stream->urb_list[i]->transfer_flags = 0;
+		stream->urb_list[i]->transfer_flags = URB_NO_TRANSFER_DMA_MAP;
+		stream->urb_list[i]->transfer_dma = stream->dma_addr[i];
 		stream->urbs_initialized++;
 	}
 	return 0;

